### PR TITLE
Add the ability to specify what repos to include/exclude while building 

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Add the ability to include/exclude certain repos when building universe

cc @pranavkm @BrennanConroy @troydai

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -67,8 +29,8 @@ functions
 
 var buildTarget = "compile"
 
-@{
-   var kBuildVersion = Environment.GetEnvironmentVariable("DNX_BUILD_VERSION");
+@{
```
Comment (BrennanConroy) [inline makefile.shade]: spaces

**Comment by BrennanConroy** [comment]
Looks good to me

File: makefile.shade [hunk 2]
```diff
@@ -699,4 +661,64 @@ functions
...
+                "WebListener",
+                "KestrelHttpServer",
+                "WebSockets",
+                "Session",
+                "UserSecrets",
+                "CORS",
+                "Entropy",
+            };
+            
+            var repositoryInclude = Environment.GetEnvironmentVariable("REPOSITORY_INCLUDE");
```
Comment (pranavkm) [inline makefile.shade]: `DNX_REPOSITORY_INCLUDE` \ `KOREBUILD_REPOSITORY_INCLUDE`?

File: makefile.shade [hunk 3]
```diff
@@ -699,4 +661,64 @@ functions
 
             return true;
         }
+        
+        static IEnumerable<string> GetRepositoriesToBuild() 
+        {
+            IEnumerable<string> reposToBuild = new HashSet<string>(StringComparer.OrdinalIgnoreCase) {
```
Comment (pranavkm) [inline makefile.shade]: `var`

**Comment by Eilon** [comment]
Yeah :shipit: very nice.

**Comment by pranavkm** [comment]
:shipit:

File: makefile.shade (hunk 3 above)
Comment (victorhurdugaci) [inline makefile.shade]: var doesn't work well here. I am assigning an IEnumerable later but the first time the type is resolved as HashSet. I would have to cast it anyway...

File: makefile.shade (hunk 2 above)
Comment (victorhurdugaci) [inline makefile.shade]: KOREBUILD it is

**Pull request merged**